### DISCUSSION TITLE
Get `omnisharp-fix-usings` in eglot?

### DISCUSSION THREAD

**Discussion by Kevinizevbigie** [joaotavora/eglot#1426] [Q&A]
I tested the `omnisharp` package with Emacs which has a nice function I like called `omnisharp-fix-usings`. 

But I use Eglot for everything else, so it would be better if I had Eglot fully functional with C#.

Is there a way to make this functionality work with Eglot?

### For example: 

`ReadStdin` is in the hello package

![1](https://github.com/user-attachments/assets/75a3b615-77b5-422e-8a0a-6a3ad3116914)

After `omnisharp-fix-usings`, `using hello` is identified and added.

![2](https://github.com/user-attachments/assets/a38460f6-a630-438e-a712-de0db1a4bafa)

**Comment by joaotavora** [comment]
If the LSP supports code actions, maybe `M-x eglot-code-action-organize-imports` works, or at least does something similar.

**Comment by Kevinizevbigie** [comment]
I tried various scenarios with `M-x eglot-code-action-organize-imports`.

The message I get is the same.

![2024-07-24_11-22](https://github.com/user-attachments/assets/2b9a06d1-e3d2-4f45-b116-39ae09cf48da)

**Comment by Kevinizevbigie** [comment]
After looking into this more, It seems that `M-x eglot-code-action` or `M-x eglot-code-action-quickfix` creates a list of actions which include the using statement.

But it only works if the cursor in on the error related to the using statement.

![2024-09-26_00-33](https://github.com/user-attachments/assets/b3bbde46-288d-4a2a-ba4e-a4ffd22fca3b)
